oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/client/InMemoryRegisteredClientRepository.java

```diff
@@ -103,6 +103,12 @@ private void assertUniqueIdentifiers(RegisteredClient registeredClient, Map<Stri
 				throw new IllegalArgumentException("Registered client must be unique. " +
 						"Found duplicate client identifier: " + registeredClient.getClientId());
 			}
+
+			// client_secert: OAuth 2.0 client secret string.
```

Comment:
I don't feel this comment is necessary as the exception message - "Registered client must be unique. Found duplicate client secret for identifier: " - clearly indicates that the client secret must be unique.